synapse/events/snapshot.py

```diff
@@ -198,6 +201,11 @@ def deserialize(storage: "Storage", input: JsonDict) -> "EventContext":
             rejected=input["rejected"],
         )
 
+        if input["outlier"]:
+            context._prev_state_ids = {}
+            context._current_state_ids = {}
+            context._outlier = True
```

Comment:
Going down this route, it seems like we could short-circuit and return early for an outlier and just create `EventContext.for_outlier()`:

```py
if input["outlier"]:
  return EventContext.for_outlier()
```

Reply:
Other metadata can still be added for outliers (like the appservice metadata below this)

Reply:
I see 👍